File: Services/hthservices/hthservicesWeb/hthservices/Ads/AdItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace hthservices.Ads
{
    public class AdItem
    {
        public String NameVN;
        public String Name;
        public String DescVN;
        public String Desc;
        public String Type;
        public String Link;
        public String AppId;
        public String UrlImage;

        /*---------- For owner ads --------- */

        public string InCountries; //VN, EN, 

        public string IgnoreCountries; //VN, EN, 

        public string GetName(string country)
        {
            if ("VN".Equals(country, StringComparison.OrdinalIgnoreCase))
            {
                if (!string.IsNullOrWhiteSpace(NameVN)) return NameVN;
            }
            return Name;
        }

        public string GetDesc(string country)
        {
            if ("VN".Equals(country, StringComparison.OrdinalIgnoreCase))
            {
                if (!string.IsNullOrWhiteSpace(DescVN)) return DescVN;
            }
            return Desc;
        }
        public bool IsAllowForCountry(string country)
        {
            if (!string.IsNullOrWhiteSpace(country))
            {
                string countryInFormat = country.ToLower() + ",";
                if (!string.IsNullOrWhiteSpace(InCountries))
                {
                    string inCountriesInformat = InCountries.Trim().ToLower() + ",";
                    if (inCountriesInformat.ToLower().Contains(countryInFormat.ToLower())) { return true; }
                    else { return false; }
                }
                if (!string.IsNullOrWhiteSpace(IgnoreCountries))
                {
                    string ignoreCountries = IgnoreCountries.Trim().ToLower() + ",";
                    if (ignoreCountries.ToLower().Contains(countryInFormat.ToLower())) { return false; }
                    else { return true; }
                }
            }
            return true;
        }

        private string packageName = null;
        [IgnoreDataMember]
        public string PackageName
        {
            get
            {
                if (packageName == null)
                {
                    if (!String.IsNullOrWhiteSpace(Link))
                    {
                        var start = Link.LastIndexOf("=", StringComparison.OrdinalIgnoreCase);
                        if (start > 0)
                        {
                            packageName = Link.Substring(start + 1).Trim().ToLower();
                        }
                    }
                    else
                    {
                        packageName = String.Empty;
                    }
                }
                return packageName;
            }
        }
    }
}

File: tinchungkhoan/app/src/main/java/com/hunght/utils/AdItem.java
package com.hunght.utils;

public class AdItem {

	public String Name;
	public String Desc;
	public String Type;
	public String Link;
	public String AppId;
	public String UrlImage;

        /*---------- For owner ads --------- */

	public String InCountries; //VN, EN,

	public String IgnoreCountries; //VN, EN,

	public String getName() {
		if(Name == null){
			return "";
		}
		else{
			return Name;
		}
	}

	public void setName(String name) {
		Name = name;
	}

	public String getDesc() {
		if(Desc == null){
			return "";
		}
		else{
			return Desc;
		}
	}

	public void setDesc(String desc) {
		Desc = desc;
	}

	public String getType() {
		if(Type == null){
			return "";
		}
		else{
			return Type;
		}
	}

	public void setType(String type) {
		Type = type;
	}

	public String getLink() {
		if(Link == null){
			return "";
		}
		else{
			return Link;
		}
	}

	public void setLink(String link) {
		Link = link;
	}

	public String getAppId() {
		if(AppId == null){
			return "";
		}
		else{
			return AppId;
		}
	}

	public void setAppId(String appId) {
		AppId = appId;
	}

	public String getUrlImage() {
		if(UrlImage == null){
			return "";
		}
		else{
			return UrlImage;
		}
	}

	public void setUrlImage(String urlImage) {
		UrlImage = urlImage;
	}
}
